packages/vscode-extension/src/debug/localTelemetryReporter.ts

```diff
@@ -117,3 +138,113 @@ export async function sendDebugAllEvent(
     );
   }
 }
+
+export const UnknownPlaceholder = "<unknown>";
+export const UndefinedPlaceholder = "<undefined>";
+export const DefaultPlaceholder = "<default>";
+
+export function maskValue(
+  value: string | undefined,
+  knownValues: (string | { value: string; mask: string })[] = []
+): string {
+  if (typeof value === "undefined") {
+    return UndefinedPlaceholder;
+  }
+  const findValue = knownValues.find((v) =>
+    typeof v === "string" ? v === value : v.value === value
+  );
+
+  if (typeof findValue === "undefined") {
+    return UnknownPlaceholder;
+  } else if (typeof findValue === "string") {
+    return findValue;
+  } else {
+    return findValue.mask;
+  }
+}
+
+export function maskArrayValue(
+  valueArr: string[] | undefined,
+  knownValues: (string | { value: string; mask: string })[] = []
+): string[] | string {
+  if (typeof valueArr === "undefined") {
+    return UndefinedPlaceholder;
+  }
+
+  return valueArr.map((v) => maskValue(`${v}`, knownValues));
+}
+
+interface ITaskJson {
+  tasks?: ITask[];
+}
+
+interface ITask {
+  label?: string;
+  type?: string;
+  command?: string;
+  dependsOn?: string | string[];
+}
+
+interface IDependsOn {
+  label: string;
+  type: string;
+  command: string;
+}
+
+interface IPreLaunchTaskInfo {
+  m365Overall?: IDependsOn[];
+  overall?: IDependsOn[];
+}
+
+export async function getPreLaunchTaskInfo(): Promise<IPreLaunchTaskInfo | undefined> {
```

Comment:
Add tests to cover:
- happy path
- error reading tasks.json
- user data is masked correctly

Reply:
Added ut for `getPreLaunchTaskInfo`